### PR TITLE
sys.argv added for cli command, test written and pass

### PR DESCRIPTION
**Pull request by ademhilmibozkurt**
I add `sys.argv` for your opened issue #38 and wrote a test `test_create_wargs`. If something wrong please warn me.

### CODE REVIEW

File: teachbooks/serve.py [hunk 1]
```diff
@@ -46,7 +46,12 @@ def __init__(self, servedir: Path | str, workdir: Path | str, port: int | None =
             old_instance = self.load(workdir)
             self.__dict__ = old_instance.__dict__
         except ServerError:
-            self.servedir = Path(servedir)
+            # Check is arguments were given
+            if len(sys.argv) == 2:
```
Comment (rlanzafame) [inline teachbooks/serve.py:50]: i don't understand where these arguments would come from, nor why the second would be the server directory.

File: tests/test_server.py [hunk 2]
```diff
@@ -1,14 +1,18 @@
 import os
+import sys
 from pathlib import Path
 
 import pytest
+from pytest import MonkeyPatch
```
Comment (rlanzafame) [inline tests/test_server.py:6]: where is this used? the package is new to me,.

File: tests/test_server.py [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 from flaky import flaky
 
 from teachbooks.serve import Server
 
 SERVE_DIR = Path(".")
 WORK_DIR = Path("./.teachbooks")
 
+sys.argv = ['serve.py', './']
```
Comment (rlanzafame) [inline tests/test_server.py:14]: is your assumption that we would be able to pass the path as an argument when calling serve.py from a CLI?

**Review by rlanzafame — COMMENTED** [review]
thank you for the PR @ademhilmibozkurt ! 
Unfortunately I don't see how this fits in the teachbooks package usage (especailly as a CLI tool with click). Unless I missed something, I will close this PR.
For your information I have done most of the planned improvements to the CLI `serve` command and will merge branch `server-fix` into `dev` soon with PR #46

**Comment by rlanzafame** [comment]
I see the PR is closed because the branch is deleted. If you illustrate why the PR is needed I can reassign it into `dev`

File: tests/test_server.py (hunk 2 above)
Comment (ademhilmibozkurt) [reply tests/test_server.py:6]: MonkeyPatch a pytest module that using for patcing mock variables and functions. I did not use it and forgot to delete it

File: tests/test_server.py (hunk 3 above)
Comment (ademhilmibozkurt) [reply tests/test_server.py:14]: Yes, this is the test scenario for sys.argv

**Comment by ademhilmibozkurt** [comment]
Thanks your interest @rlanzafame. I hope will help you somewhere

File: teachbooks/serve.py (hunk 1 above)
Comment (ademhilmibozkurt) [reply teachbooks/serve.py:50]: I import sys. And if you import sys then you can reach argv directly

**Pull request closed without merging**